tests/unit_tests/test_validator.py

```diff
@@ -0,0 +1,190 @@
+                "type_of_place": "small bridge",
+                "UUID": "dattarro",
+            },
+        ],
+        "obligatory_fields": ["position", "name", "accessible_by", "type_of_place", "UUID"],
+        "categories": {
+            "accessible_by": ["bikes", "cars", "pedestrians"],
+            "type_of_place": ["big bridge", "small bridge"],
+        },
+        "visible_data": ["accessible_by", "type_of_place"],
+        "meta_data": ["UUID"],
+    },
+    "site_content": {},
+    "plugins": [],
+}
+
+invalid_category_value_msg = [
+    (
+        "invalid category value",
+        {
+            "name": "Grunwaldzki",
+            "position": [51.1095, 17.0525],
+            "accessible_by": ["pedestrians", "cars"],
+            "type_of_place": "vacuum cleaners shop",
+            "UUID": "hidden",
+        },
+        "type_of_place",
+    ),
+    (
+        "invalid category value",
+        {
+            "name": "Zwierzyniecka",
+            "position": [51.10655, 17.0555],
+            "accessible_by": ["bikes", "penguins"],
+            "type_of_place": "small bridge",
+            "UUID": "dattarro",
+        },
+        "accessible_by",
+    ),
+]
+
+null_values_data = {
+    "map": {
+        "data": [
+            {
+                "name": "Grunwaldzki",
+                "position": [51.1095, 17.0525],
+                "accessible_by": ["pedestrians", "cars"],
+                "type_of_place": "big bridge",
+                "UUID": None,
+            },
+            {
+                "name": "Zwierzyniecka",
+                "position": [51.10655, 17.0555],
+                "accessible_by": ["bikes", "pedestrians"],
+                "type_of_place": "small bridge",
+                "UUID": "dattarro",
+                "website": None,
+            },
+        ],
+        "obligatory_fields": ["position", "name", "accessible_by", "type_of_place", "UUID"],
+        "categories": {
+            "accessible_by": ["bikes", "cars", "pedestrians"],
+            "type_of_place": ["big bridge", "small bridge"],
+        },
+    },
+}
+
+null_values_msg = [
+    (
+        "null value",
+        {
+            "name": "Grunwaldzki",
+            "position": [51.1095, 17.0525],
+            "accessible_by": ["pedestrians", "cars"],
+            "type_of_place": "big bridge",
+            "UUID": None,
+        },
+        "UUID",
+    ),
+    (
+        "null value",
+        {
+            "name": "Zwierzyniecka",
+            "position": [51.10655, 17.0555],
+            "accessible_by": ["bikes", "pedestrians"],
+            "type_of_place": "small bridge",
+            "UUID": "dattarro",
+            "website": None,
+        },
+        "website",
+    ),
+]
+
+
+def test_fully_valid_data():
+    assert validate_from_json(fully_valid_data) == fully_valid_msg
+
+
+def test_obligagory_fields_present():
+    assert validate_from_json(missing_obligatory_fields_data) == missing_obligatory_fields_msg
```

Comment:
probably enough would be:
`assert get_missing_fields(xyz) == missing_fileds`

Also - name of this test is wrong. You're not testing if obligatory fields are present but you check whether validator returning list of missing fileds. (or sth like that)

Reply:
Same goes for other tests

Reply:
> Also - name of this test is wrong. You're not testing if obligatory fields are present but you check whether validator returning list of missing fileds. (or sth like that)

I changed the names but now I think everything has too long names :laughing: